pennylane/tape/interfaces/autograd.py

```diff
@@ -107,16 +97,19 @@ def _update_trainable_params(self):
         self.trainable_params = trainable_params
         self._all_parameter_values = params
 
-    def get_parameters(self, trainable_only=True):  # pylint: disable=missing-function-docstring
-        self._update_trainable_params()
-        params = self._all_parameter_values
+    def get_parameters(
+        self, trainable_only=True, return_arraybox=False
+    ):  # pylint: disable=missing-function-docstring
+        params = []
+        iterator = self.trainable_params if trainable_only else self._par_info
+
+        for p_idx in iterator:
+            op = self._par_info[p_idx]["op"]
+            op_idx = self._par_info[p_idx]["p_idx"]
+            params.append(op.data[op_idx])
 
-        if trainable_only:
-            params = [
-                p
-                for idx, p in enumerate(self._all_parameter_values)
-                if idx in self.trainable_params
-            ]
+        if return_arraybox:
+            return params
 
         return autograd.builtins.list(params)

```

Comment:
When `return_arraybox=False`, is the previous behavior bugged, or does `params` look the same? Just curious because it looks like the default behavior is changing too, with `params` still being passed to `autograd.builtins.list`.

Reply:
Hey Cody, this method previously wasn't buggy - while the code here has changed, the (public) logic remains identical.

The main change is that `get_parameters()` no longer calls `self._update_trainable_params()`, due to changes in the latter method, which required a bit of a rewrite of the subsequent lines.

So the output of this method should remain the same pre- and post-change.

Reply:
Sweet, thanks!